test/extensions/transport_sockets/http_11_proxy/connect_test.cc

```diff
@@ -0,0 +1,353 @@
+#include "source/common/buffer/buffer_impl.h"
+#include "source/common/network/address_impl.h"
+#include "source/common/network/filter_state_proxy_info.h"
+#include "source/common/network/transport_socket_options_impl.h"
+#include "source/extensions/transport_sockets/http_11_proxy/connect.h"
+
+#include "test/mocks/buffer/mocks.h"
+#include "test/mocks/network/io_handle.h"
+#include "test/mocks/network/mocks.h"
+#include "test/mocks/network/transport_socket.h"
+#include "test/mocks/ssl/mocks.h"
+#include "test/test_common/environment.h"
+#include "test/test_common/network_utility.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::_;
+using testing::AnyNumber;
+using testing::ByMove;
+using testing::Const;
+using testing::InSequence;
+using testing::NiceMock;
+using testing::Return;
+using testing::ReturnNull;
+using testing::ReturnRef;
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace Http11Connect {
+namespace {
+
+class Http11ConnectTest : public testing::TestWithParam<Network::Address::IpVersion> {
+public:
+  Http11ConnectTest() {}
+  void initialize(bool no_proxy_protocol = false) {
+    std::string address_string =
+        absl::StrCat(Network::Test::getLoopbackAddressUrlString(GetParam()), ":1234");
+    auto address = Network::Utility::parseInternetAddressAndPort(address_string);
```

Comment:
Can we use the full type here? Here and elsewhere. (In the make_unique call below the type is obvious so auto is fine)

Reply:
ack